### ISSUE TITLE
how to enable jupyter debugger

### ISSUE THREAD

**Issue by huangkai31** [orchest/orchest#776]
**Describe the problem this improvement solves**
It would be great if can enable jupyter debugger by default
I have tried to 
1. in settings,  configure jupyterlab and pip install xeus-python 
2. in jupyter lab ,  install @jupyterlab/debugger
Still can't get the debugger working. 

![image](https://user-images.githubusercontent.com/1758277/157382723-2b315f0c-ec26-4ddb-a9b3-ac944572f836.png)

**Comment by astrojuanlu** [comment]
Hello @huangkai31! Can you confirm that you're using the `xpython` kernel for that notebook? In the meantime, I'll try to reproduce the steps on my instance and see if I can get it to work.

**Comment by huangkai31** [comment]
@astrojuanlu  no, it still use python3 as the kernel, can't select xeus or ipython as kernel
![image](https://user-images.githubusercontent.com/1758277/157392549-55cef066-3d70-4f5d-9bfd-8ce3abc70d2a.png)

**Comment by astrojuanlu** [comment]
I confirm that installing `xeus-python` is not enough for the kernel to appear in the selection dropdown, even though the kernel appears to be installed. Running these commands in the environment set-up script:

```
#!/bin/bash

# Install additional packages used throughout the pipeline. If
# you want more granularity, you can use different environments 
# for each step.

jupyter kernelspec list

conda install xeus-python
jupyter kernelspec list
```

Gives this output:

```
Available kernels:
  python3    /opt/conda/share/jupyter/kernels/python3
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /opt/conda

  added / updated specs:
    - xeus-python

...
Available kernels:
  python3        /opt/conda/share/jupyter/kernels/python3
  xpython        /opt/conda/share/jupyter/kernels/xpython
  xpython-raw    /opt/conda/share/jupyter/kernels/xpython-raw
```

And even though the kernel is visible from the Lab itself:

![Screenshot 2022-03-09 at 08-41-37 JupyterLab · Orchest](https://user-images.githubusercontent.com/316517/157395134-b97f40fd-a8cc-404d-89c1-7375a54726fe.png)

I'm probably missing something about how Orchest manages kernels. @yannickperrenet any ideas?

**Comment by yannickperrenet** [comment]
This issue is related to #138. 

My guess would be that installing `xeus-python` inside JupyterLab does not lead to the kernel being picked up (i.e. not shown in the dropdown when selecting a kernel) because JupyterLab asks the Jupyter Enterprise Gateway for the kernels it supports. Since the kernel is installed in JupyterLab and not in the Enterprise Gateway, the Enterprise Gateway has no knowledge of such kernel.

**Comment by ricklamers** [comment]
@huangkai31 thanks for bringing this up by the way. We’ll put together instructions on how to get this to work soon that will be a manual solution for the time being. We intend to support it better “out of the box” in the future.

If you’d like to tinker some more on your own, something to keep in mind that a single environment maps to a single kernel in the Jupyter Enterprise Gateway and hence to JupyterLab. So the essence of the approach would be to make sure the default kernel that’s started in the environment (since we start the kernel through Jupyter Enterprise Gateway that’s whichever kernel is initialized by its bootscript: [bootstrap-kernel.sh](https://github.com/jupyter-server/enterprise_gateway/blob/master/etc/kernel-launchers/bootstrap/bootstrap-kernel.sh) and then [launch_ipykernel.py](https://github.com/jupyter-server/enterprise_gateway/blob/d0681051cbffd58cfb4243b17e38a9547fe88d2c/etc/kernel-launchers/python/scripts/launch_ipykernel.py#L382)) supports debugging.
